### PR TITLE
Fix matching etypes by replacing with fullmatch

### PR DESCRIPTION
**Pull request by wvangeit**

### CODE REVIEW

File: bluepymm/prepare_combos/parse_files.py [hunk 1]
```diff
@@ -174,9 +183,9 @@ def read_records():
             emodel = etype_map['mm_recipe']
             for layer in etype_map['layer']:
                 for fullmtype in fullmtypes:
-                    if mtype_regex.match(fullmtype):
+                    if fullmatch(mtype_regex, fullmtype):
```
Comment (mgeplf) [inline bluepymm/prepare_combos/parse_files.py]: why not just make sure the regex ends in '$'?

Comment (wvangeit) [reply bluepymm/prepare_combos/parse_files.py]: that's another option, but it requires e.g. recompiling the regex

Comment (wvangeit) [reply bluepymm/prepare_combos/parse_files.py]: And one has to be careful there is no $ yet

Comment (mgeplf) [reply bluepymm/prepare_combos/parse_files.py]: yeah, I mean before the regex gets compiled: where ever they are added. `process_megate_config.py::join_regex` already does something similar

**Pull request merged**